### ISSUE TITLE
Produce Strong Named Signed Assemblies 

### ISSUE THREAD

**Issue by 4nthonylin** [nsubstitute/NSubstitute.Analyzers#102]
Can the analyzers assemblies be strong named signed? [docs](https://docs.microsoft.com/en-us/dotnet/framework/app-domains/how-to-sign-an-assembly-with-a-strong-name) 

Currently running `sn -vf NSubstitute.Analyzers.CSharp.dll` says that the assembly is not signed

```
Microsoft (R) .NET Framework Strong Name Utility  Version 4.0.30319.0
Copyright (c) Microsoft Corporation.  All rights reserved.

NSubstitute.Analyzers.CSharp.dll does not represent a strongly named assembly
```

**Comment by dtchepak** [comment]
Hi @4nthonylin,
Thanks for raising this. Can you let us know why you need this to be a strong named assembly?

**Comment by ImrePyhvel** [comment]
Because otherwise users would get annoying stack of warnings in their project when upgrading analyzers. For example:

> Warning	CS8032	An instance of analyzer NSubstitute.Analyzers.CSharp.DiagnosticAnalyzers.ConflictingArgumentAssignmentsAnalyzer cannot be created from C:\...\NuGet\NSubstitute.Analyzers.CSharp.1.0.11\analyzers\dotnet\cs\NSubstitute.Analyzers.CSharp.dll: Could not load type 'NSubstitute.Analyzers.CSharp.DiagnosticAnalyzers.ConflictingArgumentAssignmentsAnalyzer' from assembly 'NSubstitute.Analyzers.CSharp, Version=1.0.6.0, Culture=neutral, PublicKeyToken=null'..	

See: https://github.com/dotnet/project-system/issues/2719#issuecomment-389219218

**Comment by dtchepak** [comment]
Thanks @ImrePyhvel , that's a really useful link. 👍 

From that link there's also a reference to [xunit's experience](https://github.com/xunit/xunit/issues/1704) with this.

Would you be happy with this @tpodolak? We can use a not-so-secret key like xunit.

**Comment by tpodolak** [comment]
Closed by #125
